### PR TITLE
chore: upgrade to unity 2021.3.20f1

### PR DESCRIPTION
**Pull request by 0xD-Fabio**
## What does this PR change?

This PR is meant to upgrade Unity version to 2021.3.20f1, which will fix our [crash on exit problem](https://app.zenhub.com/workspaces/workstream-core-experience-63aaac346e1c8d00104586e8/issues/gh/decentraland/unity-renderer/3665)

One caveat on this upgrade is a temporary bypass of a test failure due to a bug introduced in Unity 2021.3.17f1.
This bug has been investigated in the ticket opened to the Unity support team in [this thread](https://support.unity.com/hc/en-us/requests/1483408?page=1).
Here they suggested temporarily commenting out our test `public IEnumerator FailWithBogusTexture()` until they fix this known issue on their side.

## How to test the changes?

Changes include some texture manipulation functions.
The best way to test would be to teleport to crowded areas and see if the loaded texture behave normally.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Comment by 0xD-Fabio** [comment]
NOTE: Unity has introduced a bug from version 2021.3.17f1 ([see thread](https://support.unity.com/hc/en-us/requests/1483408?page=1)).
We need to fix it with some workaround, then update the CI docker image and update `browser-interface/static/unity.loader.js` with the correct versions.

**Review by eordano — APPROVED** [review]

**Review by kuruk-mm — APPROVED** [review]

File: unity-renderer/ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -17,7 +17,7 @@ PlayerSettings:
   defaultCursor: {fileID: 0}
   cursorHotspot: {x: 0, y: 0}
   m_SplashScreenBackgroundColor: {r: 1, g: 1, b: 1, a: 1}
-  m_ShowUnitySplashScreen: 0
+  m_ShowUnitySplashScreen: 1
```
Comment (kuruk-mm) [inline unity-renderer/ProjectSettings/ProjectSettings.asset:20]:
```suggestion
  m_ShowUnitySplashScreen: 0
```

Can you revert this file?

**Review by kuruk-mm — CHANGES_REQUESTED** [review]

File: unity-renderer/ProjectSettings/ProjectSettings.asset (hunk 1 above)
Comment (0xD-Fabio) [reply unity-renderer/ProjectSettings/ProjectSettings.asset]: reverted ProjectSettings.asset file as it was before

**Review by kwsskulinski — APPROVED** [review]
Textures behave normally without any issues.

**Review by kwssbocian — APPROVED** [review]
✔️No issues observed regarding texture appearance or crashes.

**Review by Kinerius — APPROVED** [review]
LGTM!

**Review by sandrade-dcl — APPROVED** [review]

**Pull request merged**